Document merging guidelines (#73)
jupyter-book/team-compass: docs/practices.md

These are team practices that we try to follow across all repositories of `jupyter-book`.
They are not _strictly_ enforced, but we ask that you prioritize following them, and provide a clear rationale if you don't intend to.

## When to merge pull requests

> We want to encourage fluid contributions and a bias towards action, rather than worrying about getting it perfect.

Reviewer: I like the re-centering on iteration here, as that is the way to get towards quality.

> We aim for optimistic merges, with [core-team contributors](./team.md) using their best judgement of when they think something might be controversial. Here are a few guidelines to follow:

Reviewer 2: ```suggestion
We aim for optimistic merges, with [core-team contributors](./team.md) using their best judgement to determine whether a change might be controversial. Here are a few guidelines to follow:
```

- **Aim to merge only changes you understand fully**, if not seek review from the rest of the team.
- **Use the [`#development` channel](https://discord.com/channels/1083088970059096114/1177359316886503464) in Discord to ask questions**. That's a useful place to signal-boost questions and discussion.
- **Smaller, iterative changes**: are low risk - aim to seek feedback, but don't block yourself if you're not getting a response! 
    - Examples: targeted bug fixes, typos, docs changes.
- **Larger changes for maintainers or users**: more important to get review - seek input and post messages in both GitHub and Discord.
    - Examples: An AST change, a change to core processing infrastructure, a change to team practices everybody will need to follow, a new directive, a breaking change for users we can't easily walk back. 
- **Use the GitHub UI to review your own pull request**! If you'd like to merge without another person reviewing first, this is a helpful way to show you've done some due-diligence.

> In general, we trust those with merge rights to make the right decision. You should use your best judgment and it's OK if you make a mistake - we can always revert changes if we haven't made a release.

Reviewer 3: sometimes it's after release
```suggestion
In general, we trust those with merge rights to make the right decision. You should use your best judgment and it's OK if you make a mistake - we can always revert changes if it's necessary.
```
